### PR TITLE
webContents: adding events to detect gpu and plugin process crashes

### LINKED ISSUE #832: Detecting GPU process crashes
I'm trying to make my app restart itself if any of the renderer processes crash. So far, I've been successful, except that if the GPU process crashes (for testing purposes, by me `TerminateProcess`'ing it), my app turns to a white screen until I call `browserWindow.webContents.refresh()`. 

While this error appears recoverable, I just need an event on `app` to tell me when the GPU process has died, so that I can do the refresh.

### PR DESCRIPTION
**Pull request by deepak1556**
Fixes #832 , thought it would be good to have event to detect plugin crashes too. Thoughts ?

### CODE REVIEW

File: atom/browser/api/atom_api_web_contents.cc [hunk 1]
```diff
@@ -225,6 +226,20 @@ void WebContents::RenderProcessGone(base::TerminationStatus status) {
...
+                                base::ProcessId plugin_pid) {
+  content::WebPluginInfo info;
+  auto plugin_service = content::PluginService::GetInstance();
+  plugin_service->GetPluginInfoByPath(plugin_path, &info);
+  Emit("plugin-crashed", info.name, info.version);
+}
+
+void WebContents::OnGpuProcessCrashed(base::TerminationStatus exit_code) {
+  auto status = base::TerminationStatus::TERMINATION_STATUS_PROCESS_CRASHED;
+  if (exit_code == status)
```
Comment (zcbenz) [inline atom/browser/api/atom_api_web_contents.cc]: We can use `exit_code == base::TERMINATION_STATUS_PROCESS_CRASHED` directly.

Comment (deepak1556) [inline atom/browser/api/atom_api_web_contents.cc]: Line length >= 80

Comment (zcbenz) [inline atom/browser/api/atom_api_web_contents.cc]: Without the `TerminationStatus::` prefix line length is <= 80.

Comment (deepak1556) [inline atom/browser/api/atom_api_web_contents.cc]: oops sorry, totally missed that. Fixed, thanks!

**Comment by zcbenz** [comment]
:+1:

**Pull request merged**